src/mds/MDCache.cc

```diff
@@ -8358,6 +8358,40 @@ int MDCache::path_traverse(const MDRequestRef& mdr, MDSContextFactory& cf,
   if (flags & MDS_TRAVERSE_CHECK_LOCKCACHE)
     mds->locker->find_and_attach_lock_cache(mdr, cur);
 
+  if (mdr) {
+    const cref_t<MClientRequest> &req = mdr->client_request;
+    Session *session = mds->get_session(req);
```

Comment:
I'm seeing a MDS crash here:

```
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr:ceph-mds: /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos9/DIST/centos9/MACHINE_SIZE/gigantic/release/19.0.0-5166-g2dde9662/rpm/el9/BUILD/ceph-1
9.0.0-5166-g2dde9662/redhat-linux-build/boost/include/boost/smart_ptr/intrusive_ptr.hpp:201: T* boost::intrusive_ptr<T>::operator->() const [with T = const Message]: Assertion `px != 0' failed.
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr:*** Caught signal (Aborted) **
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: in thread 7f1882a00640 thread_name:MR_Finisher
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: ceph version 19.0.0-5166-g2dde9662 (2dde96629934037105be962838f7de476059de3a) squid (dev)
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 1: /lib64/libc.so.6(+0x3e6f0) [0x7f189123e6f0]
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 2: /lib64/libc.so.6(+0x8b94c) [0x7f189128b94c]
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 3: raise()
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 4: abort()
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 5: /lib64/libc.so.6(+0x2871b) [0x7f189122871b]
2024-07-20T05:05:31.195 INFO:tasks.ceph.mds.c.smithi031.stderr: 6: /lib64/libc.so.6(+0x37386) [0x7f1891237386]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 7: (MDSRank::get_session(boost::intrusive_ptr<Message const> const&)+0x89) [0x5565b16c36b9]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 8: (MDCache::path_traverse(boost::intrusive_ptr<MDRequestImpl> const&, ContextFactory<MDSContext>&, filepath const&, int, std::vector<CDentry*, std::allocator<CDentry*> >*, CInode**, CDir**)+0x235) [0x5565b
188147d]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 9: (MDCache::enqueue_scrub_work(boost::intrusive_ptr<MDRequestImpl> const&)+0xa6) [0x5565b18aca32]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 10: (MDCache::enqueue_scrub(std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, bool, bool, bool, bool, ceph::Formatter*, Context*)+0x966) [0x5565b1
8ad4da]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 11: (MDSRank::command_scrub_start(ceph::Formatter*, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::vector<std::__cxx11::basic_string<char,
 std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, Context*)+0x234) [0x5565b16d6556]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 12: ceph-mds(+0x15817e) [0x5565b170717e]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 13: ceph-mds(+0xeb3b1) [0x5565b169a3b1]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 14: (Finisher::finisher_thread_entry()+0x495) [0x7f1891bd45a5]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 15: ceph-mds(+0x103399) [0x5565b16b2399]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 16: (Thread::entry_wrapper()+0x43) [0x7f1891c25ced]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 17: (Thread::_entry_func(void*)+0xd) [0x7f1891c25d09]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 18: /lib64/libc.so.6(+0x89c02) [0x7f1891289c02]
2024-07-20T05:05:31.196 INFO:tasks.ceph.mds.c.smithi031.stderr: 19: /lib64/libc.so.6(+0x10ec40) [0x7f189130ec40]
2024-07-20T05:05:31.197 INFO:tasks.ceph.mds.c.smithi031.stderr:2024-07-20T05:05:31.194+0000 7f1882a00640 -1 *** Caught signal (Aborted) **
2024-07-20T05:05:31.197 INFO:tasks.ceph.mds.c.smithi031.stderr: in thread 7f1882a00640 thread_name:MR_Finisher
```
[
[vshankar-2024-07-19_17:58:44-fs-wip-vshankar-testing-20240719.093430-debug-testing-default-smithi](https://pulpito.ceph.com/vshankar-2024-07-19_17:58:44-fs-wip-vshankar-testing-20240719.093430-debug-testing-default-smithi/)/7808510](https://pulpito.ceph.com/vshankar-2024-07-19_17:58:44-fs-wip-vshankar-testing-20240719.093430-debug-testing-default-smithi/7808510/)

Reply:
This is triggered via scrub, so, I think the `Session*` is nullptr.